cleo_sdk/CLEO_Utils.h

```diff
@@ -112,34 +100,123 @@ namespace CLEO
         return result;
     }
 
+    static bool StringStartsWith(const std::string_view str, const std::string_view prefix, bool caseSensitive = true)
+    {
+        if (str.length() < prefix.length())
+        {
+            return false;
+        }
+
+        if (caseSensitive)
+        {
+            return strncmp(str.data(), prefix.data(), prefix.length()) == 0;
+        }
+        else
+        {
+            return _strnicmp(str.data(), prefix.data(), prefix.length()) == 0;
+        }
+    }
+
     static std::string ScriptInfoStr(CLEO::CRunningScript* thread)
     {
         std::string info(1024, '\0');
         CLEO_GetScriptInfoStr(thread, true, info.data(), info.length());
         return std::move(info);
     }
 
-    // does file path points inside game directories? (game root or user files)
+    // Normalize filepath, collapse all parent directory references. Input should be absolute path without expandable %variables%
+    static void NormalizeFilepath(std::string& path, bool normalizeCase = true)
+    {
+        if (path.empty()) return;
+
+        std::replace(path.begin(), path.end(), '/', '\\');
+        if (normalizeCase) std::transform(path.begin(), path.end(), path.begin(), [](unsigned char c) { return tolower(c); }); // to lower case
+
+        // collapse references to parent directory
+        const auto ParentRef = "\\..\\";
+        const auto ParentRefLen = 4;
+
+        size_t refPos = path.find(ParentRef);
+        while (refPos != std::string::npos && refPos > 0)
+        {
+            size_t parentPos = path.rfind('\\', refPos - 1); // find start of parent name
+
+            if (parentPos == std::string::npos)
+                return; // parent must be root of the path then. We want to keep absolute path, let it be as is (even if "C:\..\" makes no sense)
+
+            path.replace(parentPos, (refPos - parentPos) + ParentRefLen - 1, ""); // remove parent and parent reference
+
+            refPos = path.find(ParentRef); // find next
+        }
+
+        while(path.back() == '\\') path.pop_back(); // remove trailing path separator(s)
+    }
+
+    static std::string GetGameDirectory() // already stored in Filepath_Game
+    {
+        static const auto GTA_GetCWD = (char* (__cdecl*)(char*, int))0x00836E91; // SA 1.0 US ingame function
```

Comment:
please move it CGameVersionManager, it's better to keep all hooks in one place for future support of more versions/games.

can be done in a follow-up PR

Reply:
This is CLEO_Utils from SDK. Means to be minimal as possible header only.

Reply:
Yes. Still, in the long run we don't want to hardcode things here.

Reply:
What is solution then? Creating new export "get_addres_of" and big enum with available options?

Reply:
not sure yet. need to come up with a generic solution that will for in III/VC as well

Reply:
Would be better if plugins in some way could get pointers to these two strings in main CLEO, instead of everybody creating their own instance.

Reply:
CLEO_GetGameDir
CLEO_GetUserDir

Reply:
CLEO Redux, for example, provides GetCwd export
https://github.com/cleolibrary/CLEO-Redux/blob/master/SDK/cleo_redux_sdk.h#L75C7-L75C13

Reply:
CLEO_ResolvePath with `cleo:` or `user:`

Reply:
created a ticket to track this